common/txmgr/confirmer.go

```diff
@@ -605,10 +627,12 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Reb
 
 	// It is safe to process separate keys concurrently
 	// NOTE: This design will block one key if another takes a really long time to execute
-	wg.Add(len(ec.enabledAddresses))
+	rebroadcastAddrs := ec.enabledAddresses
+	rebroadcastAddrs = append(rebroadcastAddrs, ec.tracker.GetAbandonedAddresses()...)
+	wg.Add(len(rebroadcastAddrs))
 	errors := []error{}
 	var errMu sync.Mutex
-	for _, address := range ec.enabledAddresses {
+	for _, address := range rebroadcastAddrs {
```

Comment:
We may not want to include abandoned transaction in this rebroadcast method. I think rebroadcasting here consists of fixing errors like insufficient funds and gas bumping. Abandoned keys likely wouldn't have funds to perform these actions

Reply:
Yes, please remove abandones Txes from rebroadcasting, because that logic tries to bump gas.
These abandoned txes will be rebroadcasted anyways by the resender.